### ISSUE TITLE
adding the conversation directly to GUI

### ISSUE THREAD

**Issue by naufalHaidar12342** [andreabradpitto/Mentore-Group-Project#3]
Hello! 

Friend of mine whose assigned to replicate the text-to-speech and speech-to-text is curious, is it possible to add the conversation function directly to GUI? It seems that the conversation in Mentore app is performed through terminal, which is less suited for our potential client who need the GUI for conversation. 

If so, what are the requirement to do it? Thanks!

**Comment by andreabradpitto** [comment]
This feature has not been implemented in our project. Currently, you can only chat via terminal, and leverage the GUI for editing your ontology. I'll explain what I would do in your place to try and achieve your goal. I do not grant that this is the best way, but it could work.

First, you can add another button and a corresponding page to the original GUI, that will let users chat directly inside the Qt app. To do so, you can start by copying one of the window pages I already designed (like the one for sentence addition), and have the user type a text input. You can do so by displaying "Insert your text here to start chatting" just like I had done with the greyed out "recent" buttons in the GUI's main screen. The new window should probably feature another box that prints the bot's answer, or something similar. Even better, you could feature a single text box that resembles common messaging apps (e.g. WhatsApp, Telegram), which is more intuitive.

Then, the text should be sent to the dialogue manager, actually taking place of [this](https://github.com/andreabradpitto/Mentore-Group-Project/blob/ec2e84aaf9cd9a3cd01169c9a22bd2e21f61ae68/Flask_CARESSES/test_dialogue_manager.py#L29) line. You then have to redirect all the meaningful inputs to the response text box (or the same text box, if you have only one as suggested before), which will show the bot's answer(s).

To sum up, I think you should have to merge the two branches into a single common folder, featuring all the necessary files for both the GUI and the chat bot. Then, you could move the interface between the GUI and the chat portion inside the new window page you created, so that you can easily capture user input and print response outputs.

I think that achieving that is relatively easy from the GUI part (you just have to add a new page and to modify it a bit), but I cannot speak for the other part, as I was not directly involved in the code. Maybe you will have to re-design some portions of that branch in other to suit your needs. Maybe studying the whole code yourselves will yield better ideas.

I hope I could help!

**Comment by naufalHaidar12342** [comment]
That is awesome idea! I'm currently trying to implement your idea for the conversation with GUI. Meanwhile, is there a setting in PyQt5 to adjust the GUI automatically to the center if in full screen mode? Or should I add another setting manually, in mainWindow.py for example? 

The target screen for our chatbot is computer monitor, around 20 inch monitor with touchscreen capability. Or maybe should we set the resolution on the monitor to match the resolution of mentore app?

**Comment by andreabradpitto** [comment]
Our GUI was tailored for the screen of the robot we wanted to use it on, so we set all the ratios accordingly.
You can specify the size of the window [here](https://github.com/andreabradpitto/Mentore-Group-Project/blob/GUI/mentore.py#L23). If you change that value, you may then have to change other values throughout the rest of the code (for instance, [here](https://github.com/andreabradpitto/Mentore-Group-Project/blob/GUI/mainWindow.py#L67)), and adjust the size of the fonts accordingly. Sadly, we did not plan our code so that these changes would be easily adjustable, as again the code is specifically designed for an 800x480 screen. It should not be that hard to fix that anyway. Maybe I suggest to try and find better solutions to organize window elements, rather than resorting to numerical values like I did. Our hard-coded numbers are the reason why, when you maximize the window, you do not have all the elements automatically centered and resized accordingly.

I do not suggest changing the resolution of the monitor to match the one of the GUI, as that would lead to a terrible user experience in my opinion.
